src/shared/tpm2-util.c

```diff
@@ -1679,9 +1891,23 @@ int tpm2_unseal(const char *device,
         if (r < 0)
                 return r;
 
-        r = tpm2_make_primary(c.esys_context, &primary, primary_alg, NULL);
-        if (r < 0)
-                return r;
+        /* If their is a primary key we trust, like an SRK, use it */
+        if (srk_buf) {
+                log_debug("Found existing SRK key to use, deserializing ESYS_TR");
+                rc = sym_Esys_TR_Deserialize(
+                                c.esys_context,
+                                srk_buf,
+                                srk_buf_size,
+                                &primary);
+                if (rc != TSS2_RC_SUCCESS)
+                        return log_error_errno(SYNTHETIC_ERRNO(ENOTRECOVERABLE),
+                                               "Failed to unmarshal primary key: %s", sym_Tss2_RC_Decode(rc));
+        /* old callers without an SRK still need to create a key */
+        } else {
+                r = tpm2_make_transient_primary(c.esys_context, &primary, primary_alg, false, NULL);
```

Comment:
i recent code we like if opaque arguments in function calls, such as "true", "false" or NULL, or 0 are prefixed with the name of the argument, since from the values you pass you otherwise have no idea what this might be.

hence maybe:

```c
r = tpm2_make_transient_primary(
                c.esys_context, 
                &primary, 
                primary_alg, 
                /* use_srk_model= */ false, 
                /* ret_arg= */ NULL);